node/src/components/chain_synchronizer/operations.rs

```diff
@@ -737,30 +739,62 @@ where
             parent: Box::new(parent_header.clone()),
         })?;
 
-    // Ensure we have at least one peer to which we're bidirectionally connected before trying to
-    // fetch the data.
-    let mut peers = vec![];
-    for _ in 0..ctx.config.max_retries_while_not_connected() {
-        peers = get_filtered_fully_connected_peers(ctx).await;
-        if !peers.is_empty() {
-            break;
+    for _ in 0..=1 {
+        let peers = prepare_peers_applicable_for_block_fetch(ctx).await;
+        let peer_count = peers.len();
+        let maybe_item = try_fetch_block_or_block_header_by_height(
+            peers.clone(),
+            ctx,
+            height,
+            parent_header,
+            key_block_info,
+        )
+        .await?;
+        match maybe_item {
+            Some(item) => {
+                check_block_version_is_not_greater_than_current(
+                    item.header(),
+                    ctx.config.protocol_version(),
+                )?;
+                check_block_version_is_not_lower_than_parent(item.header(), parent_header)?;
+                return Ok(Some(item));
+            }
+            None => {
+                if peer_count >= ctx.config.minimum_peer_count_threshold_for_fetch_retry {
+                    warn!(
+                        %height,
+                        attempts_to_get_fully_connected_peers =
+                            %ctx.config.max_retries_while_not_connected(),
+                        "unable to fetch item despite having enough peers"
+                    );
+                    break;
+                }
+                info!(
+                    %height,
+                    %peer_count,
+                    minimum_peer_count_threshold = %ctx.config.minimum_peer_count_threshold_for_fetch_retry, "tried fetching with not enough peers, may try again"
```

Comment:
Could this be manually adjusted to not exceed the 100 char limit pls?

Reply:
Updated in 0b7f912834fb322bac951d8a4c2df8fffbdc1756